lib/reverse_words.rb

```diff
@@ -1,4 +1,45 @@
 # A method to reverse each word in a sentence, in place.
+require 'pry'
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil || my_words.length == 0
+    return my_words
+  end
+  string_length = my_words.length
+  i = 0
+  while i < string_length
+    space = find_next_space_index(my_words, i)
+    if space == 0
+      j = 0
+    else j = space - 1
+    end
+    my_words[i..j] = string_reverse(my_words[i..j])
+    i = space + 1
+  end
+  return my_words
 end
+
+def find_next_space_index(string, i)
+    while string[i] !=  " " &&  i < string.length
+    i += 1
+    end
+  space = i
```

Comment:
Minor: you could simply return i and don't need to create another variable called space in this method.